allensdk/internal/brain_observatory/time_sync.py

```diff
@@ -30,25 +36,85 @@
 
 # MPE is changing keys. This isn't versioned in the file.
 VERSION_2_KEYS = {
-    "photodiode": "photodiode",
+    "photodiode": "stim_photodiode",
     "2p": "2p_vsync",
     "stimulus": "stim_vsync",
     "eye_camera": "eye_tracking",
     "behavior_camera": "behavior_monitoring",
     "acquiring": "2p_acquiring",
     "lick_sensor": "lick_sensor"
     }
+"""
+
+POSSIBLE_KEY_PAIRS = {
+        "photodiode": ("stim_photodiode", "photodiode"),
+        "eye_camera": ("cam2_exposure", "eye_tracking"),
+        "behavior_camera": ("cam1_exposure", "behavior_monitoring"),
+        "lick_sensor": ("lick_1", "lick_sensor")
+        }
+
+
+def validate_keys(key_set: Set, dictionary_with_keys: Dict):
+    """
+    Validates that all values in the dictionary_with_keys are present in
+    the list.
+
+    Args:
+        key_set: Set of key values
+        dictionary_with_keys: Dictionary where values should be contained in
+                              key_list
+
+    Returns:
+        non_matches: returns a list of values not found in the key_set but
+                    present in dictionary
+    """
+    return_keys = []
+    for key, value in dictionary_with_keys.items():
+        if key not in key_set:
+            return_keys.append(key)
+    return return_keys
 
 
 def get_keys(sync_dset):
     """Get the correct lookup for line labels.
 
     This method is fragile, but not all old data contains the full list
     of keys.
+
+    Update (4/24/2020):
```

Comment:
git comments and commit history serve this purpose. This is quite verbose. Docstrings should be minimally succinct.

Reply:
what action was taken to resolve this conversation?

Reply:
removed comment